eras/shelley/impl/src/Cardano/Ledger/Shelley/PParams.hs

```diff
@@ -457,9 +474,13 @@ shelleyCommonPParamsHKDPairsV6 ::
   PParamsHKD f era ->
   [(Key, HKD f Aeson.Value)]
 shelleyCommonPParamsHKDPairsV6 px pp =
-  [ ("decentralisationParam", hkdMap px (toJSON @UnitInterval) (pp ^. hkdDL @era @f))
-  , ("extraEntropy", hkdMap px (toJSON @Nonce) (pp ^. hkdExtraEntropyL @era @f))
+  [ ("decentralization", hkdMap px (toRationalJSON @UnitInterval) (pp ^. hkdDL @era @f))
+  , ("extraPraosEntropy", hkdMap px nonceToJSON (pp ^. hkdExtraEntropyL @era @f))
   ]
+  where
+    nonceToJSON :: Nonce -> Aeson.Value
+    nonceToJSON NeutralNonce = Aeson.Null
+    nonceToJSON (Nonce n) = toJSON n
```

Comment:
Why not fix the ToJSON/FromJSON instance for `Nonce` instead of having a custom implementation here?